src/setuptools_scm/version.py

```diff
@@ -56,33 +77,46 @@ def callable_or_entrypoint(group, callable_or_name):
         return ep.load()
 
 
-def tag_to_version(tag):
+def tag_to_version(tag, config=None):
     """
     take a tag that might be prefixed with a keyword and return only the version part
+    :param config: optional configuration object
     """
     trace("tag", tag)
-    if "+" in tag:
-        warnings.warn("tag %r will be stripped of the local component" % tag)
-        tag = tag.split("+")[0]
-    # lstrip the v because of py2/py3 differences in setuptools
-    # also required for old versions of setuptools
-    prefix_match = TAG_PREFIX.match(tag)
-    if prefix_match is not None:
-        version = prefix_match.group(1)
-    else:
-        version = tag
+
+    if not config:
+        config = Configuration()
+
+    tagdict = _parse_version_tag(tag, config)
+    if not isinstance(tagdict, dict) or not tagdict.get('version', None):
+        warnings.warn("tag %r no version found" % (tag,))
+        return None
+
+    version = tagdict['version']
     trace("version pre parse", version)
-    if VERSION_CLASS is None:
-        return version
-    version = pkg_parse_version(version)
-    trace("version", repr(version))
-    if isinstance(version, VERSION_CLASS):
-        return version
 
+    if tagdict.get('suffix', ''):
+        warnings.warn("tag %r will be stripped of its suffix '%s'" % (tag, tagdict['suffix']))
+
+    if VERSION_CLASS is not None:
+        version = pkg_parse_version(version)
+        trace("version", repr(version))
+    
+    return version
 
-def tags_to_versions(tags):
-    versions = map(tag_to_version, tags)
```

Comment:
i believe there is a neat trick to use here - 

```
from itertools import repeat
map(tag_to_version, tags, repeat(config, len(tags))
```
but im not sure that truly aids readability as much ^^

Reply:
;-) needs to be filtered for None also.